src/xtopcom/xdb/src/xdb_rocksdb.cpp

```diff
@@ -541,10 +563,21 @@ xdb::xdb_impl::xdb_impl(const int db_kinds,const std::string& db_root_dir,std::v
 
     if ((m_db_kinds & xdb_kind_no_multi_cf) == 0)
     {
-        cf_list.push_back(setup_level_style_cf("1")); //block 'cf[1]
-        cf_list.push_back(setup_level_style_cf("2")); //block 'cf[2]
-        cf_list.push_back(setup_level_style_cf("3")); //block 'cf[3]
-        cf_list.push_back(setup_level_style_cf("4")); //block 'cf[4]
+        if(DB_OPTIONS_DEFAULT == cache_type) {
+            uint64_t memory_budget = 64 * 1024 * 1024;
+            std::shared_ptr<rocksdb::Cache> block_cache = rocksdb::NewLRUCache(8 << 20);//default 8M
+            cf_list.push_back(setup_level_style_cf("1", block_cache, memory_budget)); //block 'cf[1]
```

Comment:
duplicate code, only `memory_budget, block_cache` needs to reside in the condition statement